objects/src/accounts/storage/mod.rs

```diff
@@ -1,263 +1,78 @@
-        let slots = SimpleSmt::<STORAGE_TREE_DEPTH>::with_leaves(entries)
-            .map_err(AccountError::DuplicateStorageItems)?;
-
-        // make sure the number of provide maps matches the number of map slots
-        if maps.len() != num_maps {
-            return Err(AccountError::StorageMapTooManyMaps {
-                expected: num_maps,
-                actual: maps.len(),
-            });
+    /// Returns a new instance of account storage initialized with the provided slots.
+    pub fn new(slots: &[StorageSlot]) -> Self {
+        Self {
+            slots: slots.to_vec(),
+            commitment: build_slots_commitment(&slots),
         }
-
-        Ok(Self { slots, layout, maps })
     }
 
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
     /// Returns a commitment to this storage.
-    pub fn root(&self) -> Digest {
-        self.slots.root()
+    pub fn commitment(&self) -> Digest {
+        self.commitment
     }
 
-    /// Returns an item from the storage at the specified index.
-    ///
-    /// If the item is not present in the storage, [crate::EMPTY_WORD] is returned.
-    pub fn get_item(&self, index: u8) -> Digest {
-        let item_index = NodeIndex::new(Self::STORAGE_TREE_DEPTH, index.into())
-            .expect("index is u8 - index within range");
-        self.slots.get_node(item_index).expect("index is u8 - index within range")
+    /// Returns a reference to the slots of this storage.
+    pub fn slots(&self) -> &[StorageSlot] {
+        &self.slots
     }
 
-    /// Returns a map item from the storage at the specified index.
+    /// Returns an item from the storage at the specified index.
     ///
     /// If the item is not present in the storage, [crate::EMPTY_WORD] is returned.
-    pub fn get_map_item(&self, index: u8, key: Word) -> Result<Word, AccountError> {
-        let storage_map = self.maps.get(&index).ok_or(AccountError::StorageMapNotFound(index))?;
-
-        Ok(storage_map.get_value(&Digest::from(key)))
-    }
-
-    /// Returns a reference to the Sparse Merkle Tree that backs the storage slots.
-    pub fn slots(&self) -> &SimpleSmt<STORAGE_TREE_DEPTH> {
-        &self.slots
-    }
-
-    /// Returns layout info for this storage.
-    pub fn layout(&self) -> &[StorageSlotType] {
-        &self.layout
-    }
-
-    /// Returns a commitment to the storage layout.
-    pub fn layout_commitment(&self) -> Digest {
-        layout_commitment(&self.layout)
-    }
-
-    /// Returns the storage maps for this storage.
-    pub fn maps(&self) -> &BTreeMap<u8, StorageMap> {
-        &self.maps
+    pub fn get_item(&self, index: u8) -> Digest {
+        match self.slots.get(index as usize) {
+            Some(storage_slot) => match storage_slot {
+                StorageSlot::Value(word) => word.into(),
+                StorageSlot::Map(map) => map.root(),
+            },
+            None => EMPTY_WORD.into(),
+        }
```

Comment:
A few comments here:

- `get_item()` should return a `Word`.
- `get_item()` should panic if the index is out of bounds.
- We also need to have `get_map_item()` (similar to how we had it before).

We need to have some accessors to get info about the storage. Specifically, we need to be able to learn:
  - How many slots are in the storage.
  - What types these slots have.

These could be done via separate methods or via a single `layout()` method (similar to what we had before).